src/terraform_gui.cpp

```diff
@@ -96,6 +98,58 @@ static void GenerateRockyArea(TileIndex end, TileIndex start)
 	if (success && _settings_client.sound.confirm) SndPlayTileFx(SND_1F_SPLAT_OTHER, end);
 }
 
+struct DemolitionModeSelectionWindow : public PickerWindowBase {
+	DemolitionModeSelectionWindow(WindowDesc *desc, Window *parent) : PickerWindowBase(desc, parent)
+	{
+		this->InitNested(0);
+		this->LowerWidget(WID_DM_ALL);
+		_demolition_mode = DM_ALL;
+	}
+
+	void OnClick(Point pt, int widget, int click_count) override
+	{
+		switch (widget) {
+			case WID_DM_ALL:
+			case WID_DM_TREES:
+			case WID_DM_COMPANY_PROPS:
+				this->RaiseWidget(_demolition_mode + WID_DM_ALL);
+				_demolition_mode = (DemolitionMode)(widget - WID_DM_ALL);
+				this->LowerWidget(_demolition_mode + WID_DM_ALL);
+				if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
+				this->SetDirty();
+				break;
+		}
+	}
+};
+
+/** Nested widget definition of the demolition mode selection window */
+static const NWidgetPart _nested_demolition_mode_selection_widgets[] = {
+	NWidget(NWID_HORIZONTAL),
+		NWidget(WWT_CLOSEBOX, COLOUR_DARK_GREEN),
+		NWidget(WWT_PANEL, COLOUR_DARK_GREEN), EndContainer(),
+	EndContainer(),
+	NWidget(NWID_HORIZONTAL),
+		NWidget(WWT_IMGBTN, COLOUR_DARK_GREEN, WID_DM_ALL), SetMinimalSize(22, 22),
+								SetFill(0, 1), SetDataTip(SPR_IMG_DYNAMITE, STR_DEMOLISH_EVERYTHING_TOOLTIP),
```

Comment:
too much indentation here (though I'm not sure what the "expected" amount is, only that 4 tabs is too much)

Reply:
I am also not sure how much is expected, but I checked that file when I wrote it, and around line 370 there are the same amount of tabs where the widget attributes are in multiple lines, so I wanted to be consistent with them.